src/Components/Panels/Inventory/ObstacleSummary.tsx

```diff
@@ -78,13 +93,76 @@ function getSummary(obstacle: IBaseObject): JSX.Element {
 
 const ObstacleSummary = forwardRef<HTMLDivElement, Props>(
   ({ obstacle, onClick, selected }: Props, ref) => {
+    const [world, setBZWDocument] = useRecoilState(documentState);
+    const selectedUUID = useRecoilValue(selectionState);
+    const [nameEdit, setNameEdit] = useState(obstacle.name ?? '');
+    const [editMode, setEditMode] = useState(false);
+    const inputRef = useRef<HTMLInputElement>(null);
+
+    useEffect(() => {
+      if (editMode) {
+        inputRef.current!.focus();
+      } else {
+        inputRef.current!.blur();
+      }
+    }, [editMode]);
+
+    const classes = classList([
+      styles.wrapper,
+      [styles.selected, selected],
+      [styles.editMode, editMode],
+    ]);
+
+    const handleDoubleClick = () => {
+      setEditMode(true);
+    };
+
+    const handleSave = (e: FocusEvent<HTMLInputElement> | FormEvent) => {
+      e.preventDefault();
+      saveName();
+      setEditMode(false);
+    };
+
+    const handleOnNameChange = (e: SyntheticEvent<HTMLInputElement>) => {
+      setNameEdit(e.currentTarget.value);
+    };
+
+    const handleKeyDown = (e: KeyboardEvent<HTMLInputElement>) => {
+      if (e.keyCode === 27) {
+        setNameEdit(obstacle.name);
```

Comment:
```suggestion
        setNameEdit(obstacle.name ?? '');
```

Because `obstacle.name` can be null, if the `<input>` tag below receives a null value, you'll get this warning thrown in the console. So default to `''` just like we're doing for `nameEdit`.

```
index.js:1 Warning: A component is changing a controlled input of type text to be uncontrolled. Input elements should not switch from controlled to uncontrolled (or vice versa). Decide between using a controlled or uncontrolled input element for the lifetime of the component. More info: https://fb.me/react-controlled-components
    in input (at ObstacleSummary.tsx:156)
    in form (at ObstacleSummary.tsx:155)
    in div (at ObstacleSummary.tsx:148)
    in ForwardRef (at InventoryPanel.tsx:55)
    in div (at InventoryPanel.tsx:50)
    in InventoryPanel (at App.tsx:21)
    in div (at App.tsx:20)
    in div (at App.tsx:12)
    in App (at src/index.js:11)
    in RecoilRoot (at src/index.js:10)
    in StrictMode (at src/index.js:9)
```